src/libraries/Common/src/Interop/Browser/Interop.Runtime.cs

```diff
@@ -0,0 +1,509 @@
+            }
+            return obj == null ? 0 : (int)(IntPtr)obj.Handle;
+        }
+
+        internal static int UnBindJSObject(int js_id)
+        {
+            if (bound_objects.TryGetValue(js_id, out var obj))
+            {
+                bound_objects.Remove(js_id);
+                return obj == null ? 0 : (int)(IntPtr)obj.Handle;
+            }
+
+            return 0;
+        }
+
+        internal static void UnBindJSObjectAndFree(int js_id)
+        {
+            if (bound_objects.TryGetValue(js_id, out var obj))
+            {
+                if (bound_objects[js_id] != null)
+                {
+                    //bound_objects[js_id].RawObject = null;
+                    bound_objects.Remove(js_id);
+                }
+                if (obj != null)
+                {
+                    obj.JSHandle = -1;
+                    obj.IsDisposed = true;
+                    obj.RawObject = null;
+                    obj.Handle.Free();
+                }
+            }
+
+        }
+
+
+        internal static void UnBindRawJSObjectAndFree(int gcHandle)
+        {
+
+            GCHandle h = (GCHandle)(IntPtr)gcHandle;
+            JSObject? obj = h.Target as JSObject;
+            if (obj != null && obj.RawObject != null)
+            {
+                raw_to_js.Remove(obj.RawObject);
+
+                int exception;
+                ReleaseHandle(obj.JSHandle, out exception);
+                if (exception != 0)
+                    throw new JSException($"Error releasing handle on (js-obj js '{obj.JSHandle}' mono '{(IntPtr)obj.Handle} raw '{obj.RawObject != null})");
+
+                // Calling Release Handle above only removes the reference from the JavaScript side but does not
+                // release the bridged JSObject associated with the raw object so we have to do that ourselves.
+                obj.JSHandle = -1;
+                obj.IsDisposed = true;
+                obj.RawObject = null;
+
+                obj.Handle.Free();
+            }
+
+        }
+
+        public static void FreeObject(object obj)
+        {
+            if (raw_to_js.TryGetValue(obj, out JSObject? jsobj))
+            {
+                //raw_to_js [obj].RawObject = null;
+                raw_to_js.Remove(obj);
+                if (jsobj != null)
+                {
+                    int exception;
+                    Runtime.ReleaseObject(jsobj.JSHandle, out exception);
+                    if (exception != 0)
+                        throw new JSException($"Error releasing object on (raw-obj)");
+
+                    jsobj.JSHandle = -1;
+                    jsobj.RawObject = null;
+                    jsobj.IsDisposed = true;
+                    jsobj.Handle.Free();
+                }
+
+            }
+            else
+            {
+                throw new JSException($"Error releasing object on (obj)");
+            }
+        }
+
+        internal static object CreateTaskSource(int js_id)
```

Comment:
Would it be better to move this runtime tight method into separate file with more comments?

Reply:
Just this method or all of the methods that are used from the JS environment?  If all of them then yes but just one?

Reply:
I meant all but not sure how the split will look like

Reply:
I like the idea of all the methods actually.

Reply:
Split these out into another module.